spinoso-time/src/time/chrono/build.rs

```diff
@@ -50,11 +53,56 @@ impl Time {
             offset,
         }
     }
+
+    /// Creates a new `Time` object from the `seconds` and `sub_seconds_nanos`
+    /// since the Unix EPOCH with a local offset.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use spinoso_time::Time;
+    /// let epoch = Time::at(0, 0);
+    /// let epoch_plus_1_nano = Time::at(0, 1);
+    /// ```
+    #[inline]
+    #[must_use]
+    pub fn at(seconds: i64, sub_second_nanos: i64) -> Option<Self> {
+        let offset = Offset::Local;
+
+        let overflow = sub_second_nanos / i64::from(NANOS_IN_SECOND);
+        let timestamp = seconds.checked_add(overflow)?;
+        let sub_second_nanos = sub_second_nanos % i64::from(NANOS_IN_SECOND);
+
+        // Sub-seconds are stored as a non-negative. So negative sub-seconds are
+        // handled by subtracting one full second and calculating a new sub-second value:
+        //
+        // ```console
+        // [2.6.3] > Time.at(0, -1).to_i
+        // => -1
+        // [2.6.3] > Time.at(0, -1).nsec
+        // => 999999000
+        // ```
+        let (timestamp, sub_second_nanos) = if sub_second_nanos > 0 {
+            (timestamp, sub_second_nanos)
+        } else {
+            (
+                timestamp.checked_sub(1)?,
+                i64::from(NANOS_IN_SECOND).checked_sub(sub_second_nanos.abs())?,
```

Comment:
I think there's some carry over behavior here when `sub_second_nanos` is negative and greater than 1s.

I think this does the right thing, but I think it would be easier to follow if the overflow checking was done in each branch.

LGTM for now, I'll make a followup PR.

At the very least, I think we're missing a test.

```
[2.6.3] > t = Time.at(10, -1e6); [t.sec, t.nsec]
=> [9, 0]
[2.6.3] > t = Time.at(10, -1e6 - 1); [t.sec, t.nsec]
=> [8, 999999000]
```